llvm/include/llvm/Analysis/ScalarEvolution.h

```diff
@@ -1316,6 +1316,25 @@ class ScalarEvolution {
 
     LoopGuards(ScalarEvolution &SE) : SE(SE) {}
 
+    /// Recursively collect loop guards in \p Guards, starting from
+    /// block \p Block with predecessor \p Pred. The intended starting point
+    /// is to collect from a loop header and its predecessor.
+    static void
+    collectFromBlock(ScalarEvolution &SE, ScalarEvolution::LoopGuards &Guards,
+                     const BasicBlock *Block, const BasicBlock *Pred,
+                     SmallPtrSet<const BasicBlock *, 8> &VisitedBlocks,
```

Comment:
```suggestion
                     SmallPtrSetImpl<const BasicBlock *> &VisitedBlocks,
```

I think that should work and wouldnt' tie the API to a specific size

Reply:
adopted